Mathlib/Algebra/BigOperators/Order.lean

```diff
@@ -816,3 +816,61 @@ theorem IsAbsoluteValue.map_prod [CommSemiring R] [Nontrivial R] [LinearOrderedC
 #align is_absolute_value.map_prod IsAbsoluteValue.map_prod
 
 end AbsoluteValue
+
+namespace Mathlib.Meta.Positivity
+open Qq Lean Meta Finset
+
+/-- Prove that a finset is nonempty from a local assumption or from the fact that the finset is
+`univ` in a nonempty fintype.
+
+Note that this does not do any complicated reasoning: the only ways it can prove `s` is nonempty is
+if there is a local `s.Nonempty` hypothesis or `s = (univ : Finset α)` and `Nonempty α` can be
+synthesized by TC inference.
+
+TODO: Replace by an aesop ruleset? -/
+def proveFinsetNonempty {u : Level} {α : Q(Type u)} (s : Q(Finset $α)) :
+    MetaM Q(Finset.Nonempty $s) := do
+  try
+    let .some fv ← findLocalDeclWithType? q(Finset.Nonempty $s) | failure
+    pure $ .fvar fv
+  catch _ => do
+    let ~q(@univ _ $fi) := s | failure
+    let _no ← synthInstanceQ q(Nonempty $α)
+    pure $ q(Finset.univ_nonempty (α := $α))
```

Comment:
```suggestion
    pure <| q(Finset.univ_nonempty (α := $α))
```
Please try to stick to the style guide, though I realize muscle memory can be hard to banish